### PR TITLE
Speed up packaging

### PR DESCRIPTION
**Pull request by nirvn**
This PR reworks the packaging job workflow to prevent _3_ project loads and instead rely on a single initial load. This can lead to massive speed gains on complex projects and/or projects containing remote sources that are slow.

Requires the following libqfieldsync improvements: https://github.com/opengisch/libqfieldsync/pull/68

### CODE REVIEW

**Comment by suricactus** [comment]
The decision to load the project 3 times was by design to make each step in a Job workflow independent from the rest. Now they will become tightly coupled and "load project" becomes mandatory. Do we have any specific proof how much do we gain for such an important change that is introduced here?

+1 in putting the logic in a single function, but I would prefer to still call it three times, unless clear evidence we really save 20+ seconds on this.

File: docker-qgis/entrypoint.py [hunk 1]
```diff
@@ -138,15 +133,11 @@ def _call_libqfieldsync_packager(
...
+def _extract_layer_data() -> dict:
     logger.info("Extracting QGIS project layer data…")
 
-    project_filename = str(project_filename)
     project = QgsProject.instance()
-
-    with set_bad_layer_handler(project):
-        project.read(project_filename)
-        layers_by_id: dict = get_layers_data(project)
+    layers_by_id: dict = get_layers_data(project)
```
Comment (suricactus) [inline docker-qgis/entrypoint.py:141]: This new approach is exploiting that the `BadLayerHandler` is not cleaning the bad layer handler info after the project load is done. This should not be the case and the `BadLayerHandler` might change the way it works as soon as a QGIS crash is fixed.

**Comment by nirvn** [comment]
@suricactus , the speed gains are easy to replicate. Just take a QGIS project that loads in 10 seconds, then multiply that by 3 to get the time needlessly spent re-loading it 3 times when we can load it only once.

For one of our clients - the Groupements Forestiers Quebec - , a packaging job takes ~5 minutes, each project load taking 1 minute or so on the cloud server. I can provide job IDs, but I think we already looked at one together yesterday.

**Comment by nirvn** [comment]
Avoiding the CPU cost of reloading a project three times is also not negligible, that surely reduces stress on the server infrastructure. Project loading is costly unless you have a single geopackage + an OSM basemap layers (e.g., every time you load the project, you re-create layouts, relationships, 2D and 3D map views, refetch metadata of layers for some data providers, etc.).

Here's a packaging that lasted >4 minutes: https://app.qfield.cloud/a/ggeneau/octobre_2023/jobs?id=91a62569-2a24-454a-b9eb-53ea66292374&job-preview=1#

**Comment by suricactus** [comment]
Would be great to monitor back the actual speed improvement of these jobs. I have the gut feeling it will be negligible.

**Review by suricactus — APPROVED** [review]

**Comment by suricactus** [comment]
Ready to merge.

**Pull request merged**